src/pyscheduling/SMSP/SingleMachine.py

```diff
@@ -88,75 +34,46 @@ def generate_random(cls, jobs_number: int, InstanceName: str = "",
                         alpha: float = 2.0,
                         due_time_factor: float = 0.5,
                         Gamma: float = 2.0, Smin: int = 1, Smax: int = 25):
-        """Random generation of risijCmax problem instance
-
-        Args:
-            jobs_number (int): number of jobs of the instance
-            InstanceName (str, optional): name to give to the instance. Defaults to "".
-            protocol (SingleMachine.GenerationProtocol, optional): given protocol of generation of random instances. Defaults to SingleMachine.GenerationProtocol.VALLADA.
-            law (SingleMachine.GenerationLaw, optional): probablistic law of generation. Defaults to SingleMachine.GenerationLaw.UNIFORM.
-            Wmin (int, optional): Minimal weight. Defaults to 1.
-            Wmax (int, optional): Maximal weight. Defaults to 1.
-            Pmin (int, optional): Minimal processing time. Defaults to 1.
-            Pmax (int, optional): Maximal processing time. Defaults to 100.
-            alpha (float, optional): Release time factor. Defaults to 2.0.
-            due_time_factor (float, optional): Due time factor. Defaults to 0.5.
-            Gamma (float, optional): Setup time factor. Defaults to 2.0.
-            Smin (int, optional) : Minimal setup time. Defaults to 10.
-            Smax (int, optional) : Maximal setup time. Defaults to 100.
 
-        Returns:
-            risijCmax_Instance: the randomly generated instance
-        """
         instance = cls(jobs_number, instance_name=InstanceName)
         instance.P = instance.generate_P(protocol, law, Pmin, Pmax)
```

Comment:
Shouldn't the generation be done using the generate function inside the appropriate constraint class ? 
This way we can delete the generation code in the abstract SingleInstance / ParallelInstance ...